workspaces/ui-v2/src/hooks/useEndpointsBodyChanges.ts

```diff
@@ -0,0 +1,65 @@
+import { useSpectacleQuery } from '<src>/contexts/spectacle-provider';
+import { IChanges } from '<src>/pages/changelog/IChanges';
+
+// TODO add in changes for query parameters?
+const EndpointBodyChangesQuery = `
+query X($sinceBatchCommitId: String) {
+  requests {
+    id
+    changes(sinceBatchCommitId: $sinceBatchCommitId) {
+      added
+      changed
+    }
+    responses {
+      id
+      changes(sinceBatchCommitId: $sinceBatchCommitId) {
+        added
+        changed
+      }
+    }
```

Comment:
I'm wondering whether `responses` being nested in `requests` is the right way to go here. There are very obviously no responses without requests (unless we allow for time travel 😅), but as with requests, there are multiple per status code and per content type. However, posting a request body with one content type, doesn't mean receiving the response in that content type (`Content-Type` describing the request body, `Accept` what it accepts in return). You do wonder whether such a disconnect actually is out there in the wild, so perhaps if it's massively inconvenient to have them separate, it's not worth the bother.

Reply:
Sorry for not spotting that in earlier review :(

Reply:
So this is how spectacle is currently defined - I agree it's confusing but here's how it maps
```
requests - refer to an endpoint (http method + path)
request.bodies - refer to the actual request bodies
request.responses - refer to the response tied to the endpoint
request.responses.body - refer to the response body
```
If anything, the top level item should probably be renamed `endpoint`

Reply:
Makes sense!